Python/taskqueues-backup-images/main.py

```diff
@@ -0,0 +1,158 @@
+            "date": date,
+            "api_key": NASA_API_KEY
+        },
+    )
+    if not api_resp.ok:
+        print(
+            f"Request to NASA APOD API failed with reponse {api_resp.status_code}"
+        )
+        match api_resp.status_code:
+            case 404:  # APOD not published for the day. This is fine!
+                print("No APOD today.")
+                return "No APOD today."
+            case 500:
+                raise https_fn.HttpsError(
+                    code=https_fn.FunctionsErrorCode.UNAVAILABLE,
+                    message="APOD API temporarily not available.",
+                )
+            case _:
+                raise https_fn.HttpsError(
+                    code=https_fn.FunctionsErrorCode.INTERNAL,
+                    message="Uh-oh. Something broke.",
+                )
+    apod = api_resp.json()
+    pic_url = apod["hdurl"]
+
+    print(f"Got URL {pic_url} from NASA API for date {date}. Fetching...")
+    pic_resp = requests.get(pic_url)
+    pic_type = pic_resp.headers.get("Content-Type")
+
+    print("Uploading to Cloud Storage")
+    bucket = storage.bucket(BACKUP_BUCKET)
+    ext = pathlib.PurePosixPath(urlparse(pic_url).path).suffix
+    pic_blob = bucket.blob(f"apod/{date}{ext}")
+    try:
+        pic_blob.upload_from_string(pic_resp.content, content_type=pic_type)
+    except:
+        raise https_fn.HttpsError(code=https_fn.FunctionsErrorCode.INTERNAL,
+                                  message="Uh-oh. Something broke.")
+
+    print(f"Saved {pic_url}")
+    return f"Saved {pic_url}"
+
+
+# [START v2EnqueueTasks]
+@https_fn.on_request()
+def enqueuebackuptasks(_: https_fn.Request) -> https_fn.Response:
+    tasks_client = tasks_v2.CloudTasksClient()
+    task_queue = tasks_client.queue_path(params.PROJECT_ID.value(),
+                                         SupportedRegion.US_CENTRAL1,
+                                         "backupapod")
+    target_uri = get_function_url("backupapod")
+
+    for i in range(BACKUP_COUNT):
+        batch = i // HOURLY_BATCH_SIZE
+
+        # Delay each batch by N hours
+        schedule_delay = timedelta(hours=batch)
+        schedule_time = datetime.now() + schedule_delay
+
+        backup_date = BACKUP_START_DATE + timedelta(days=i)
+        body = {"data": {"date": backup_date.isoformat()[:10]}}
+        task = tasks_v2.Task(
+            http_request={
+                "http_method": tasks_v2.HttpMethod.POST,
+                "url": target_uri,
+                "headers": {
+                    "Content-type": "application/json"
+                },
+                "body": json.dumps(body).encode(),
+            },
+            schedule_time=schedule_time,
+        )
+        tasks_client.create_task(parent=task_queue, task=task)
+
+    return https_fn.Response(status=200,
+                             response=f"Enqueued {BACKUP_COUNT} tasks")
+# [END v2EnqueueTasks]
+
+
+# [START v2GetFunctionUri]
+def get_function_url(name: str,
```

Comment:
Now that cloudfunctions.net URLs are available in gen2, we can remove this from both the Node and Python sample. But it's still being tested in CF3 so it's not worth blocking this PR on that